tests/Unit/PointwiseFunctions/AnalyticSolutions/Elasticity/Test_HalfSpaceMirror.cpp

```diff
@@ -0,0 +1,122 @@
+    "[PointwiseFunctions][Unit][Elasticity]") {
+  const size_t dim = Elasticity::Solutions::HalfSpaceMirror::dim;
+  const Elasticity::Solutions::HalfSpaceMirror check_solution{
+      0.177, 1.0,
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>{
+          // Iron: E=72, nu=0.17
+          36.36363636363637, 30.76923076923077},
+      50, 1.e-13};
+  const auto created_solution =
+      TestHelpers::test_creation<Elasticity::Solutions::HalfSpaceMirror>(
+          "BeamWidth: 0.177\n"
+          "AppliedForce: 1.0\n"
+          "Material:\n"
+          "  BulkModulus: 36.36363636363637\n"
+          "  ShearModulus: 30.76923076923077\n"
+          "IntegrationIntervals: 20\n"
+          "IntergrationTolerance: 1.e-13\n");
+  CHECK(created_solution == check_solution);
+  test_serialization(check_solution);
+
+  pypp::SetupLocalPythonEnvironment local_python_env{"PointwiseFunctions"};
+  const Elasticity::ConstitutiveRelations::IsotropicHomogeneous<dim>
+      constitutive_relation{36.36363636363637, 30.76923076923077};
+  const HalfSpaceMirrorProxy solution{0.177, 1.0, constitutive_relation, 100,
+                                      1.e-13};
+  pypp::check_with_random_values<1,
+                                 tmpl::list<Elasticity::Tags::Displacement<dim>,
+                                            Elasticity::Tags::Strain<dim>>>(
+      &HalfSpaceMirrorProxy::field_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror",
+      {"displacement", "strain"}, {{{0., 3.}}},
+      std::make_tuple(0.177, 1.0, 36.36363636363637, 30.76923076923077),
+      DataVector(5));
+  pypp::check_with_random_values<
+      1, tmpl::list<Tags::FixedSource<Elasticity::Tags::Displacement<dim>>>>(
+      &HalfSpaceMirrorProxy::source_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror", {"source"}, {{{0., 3.}}},
+      std::make_tuple(), DataVector(5));
+
+  {
+    INFO("Test elasticity system with half-space mirror");
+    // Verify that the solution numerically solves the system and that the
+    // discretization error decreases exponentially with polynomial order
+    using system = Elasticity::FirstOrderSystem<dim>;
+    const typename system::fluxes fluxes_computer{};
+    using AffineMap = domain::CoordinateMaps::Affine;
+    using AffineMap3D =
+        domain::CoordinateMaps::ProductOf3Maps<AffineMap, AffineMap, AffineMap>;
+    const domain::CoordinateMap<Frame::Logical, Frame::Inertial, AffineMap3D>
+        coord_map{{{-1., 1., 0., 0.5}, {-1., 1., 0., 0.5}, {-1., 1., 0., 0.5}}};
+    const Mesh<dim> mesh{10, Spectral::Basis::Legendre,
+                         Spectral::Quadrature::GaussLobatto};
+    const auto logical_coords = logical_coordinates(mesh);
+    const auto inertial_coords = coord_map(logical_coords);
+    FirstOrderEllipticSolutionsTestHelpers::verify_smooth_solution<system>(
+        solution, fluxes_computer, coord_map, 1.e4, 1.,
```

Comment:
This test is quite important since it checks if the numerical derivatives of the analytic variables are consistent and solve the elasticity equations. You are currently using a very high tolerance (`1e4` with exponential scale `1.`), probably to check if the test runs at all. These parameters need to be adjusted so they actually constrain the test. To do so, output the L_2 and L_inf errors of each tensor component (one at a time) in `verify_solution` and plot them logarithmically over the number of grid points per dimension. You hopefully get an exponential decay in all components and can place a straight line in the plot with parameters A and tau such that all errors are smaller than A * exp(-tau N_points). These parameters A and tau you can use instead of the `1.e4` and `1.` in this function. Here's an example when I did this for another analytic solution: https://github.com/sxs-collaboration/spectre/pull/1704#discussion_r332580612

Reply:
I have done this and arrived at this so the test passes for `l2_error` and `linf_error`. I will do it again since I change the domain_size.

Reply:
could you post the plot like this one https://github.com/sxs-collaboration/spectre/pull/1704#discussion_r332580612 here when you have it so I can take a look at the convergence? `1.e4` seems quite large.

Reply:
Will do.